v1/plugin/plugin.go

```diff
@@ -57,6 +58,47 @@ type Publisher interface {
 	Publish([]Metric, Config) error
 }
 
+// StreamCollector is a Collector that can send back metrics on it's own
+// defined interval (within configurable limits). These limits are set by the
+// SetMaxBuffer and SetMaxCollectionDuration funcs.
+type StreamCollector interface {
+	Plugin
+
+	// StreamMetrics allows the plugin to send/receive metrics on a channel
+	// Arguments are (in order):
+	//
+	// A channel for metrics into the plugin from Snap -- which
+	// are the metric types snap is requesting the plugin to collect.
+	//
+	// A channel for metrics from the plugin to Snap -- the actual
+	// collected metrics from the plugin.
+	//
+	// A channel for error strings that the library will report to snap
+	// as task errors.
+	StreamMetrics(chan []Metric, chan []Metric, chan string) error
+	// SetMaxbuffer sets the maximum number of metrics the plugin should buffer
+	// before sending metrics.
+	SetMaxBuffer(int64)
+	// SetMaxCollectionDuration sets the maximum duration between collections
+	// before metrics should be sent (i.e.5s MaxCollectionDuration means that after
+	// 5 seconds, the plugin should send whatever it has instead of waiting longer).
+	SetMaxCollectDuration(time.Duration)
```

Comment:
Yes the sending interval is decided by the plugin. The intention behind this option was to allow the task creator to specify some bounds.